exporters/otlp/common/build.gradle.kts

```diff
@@ -11,19 +11,15 @@ plugins {
 description = "OpenTelemetry Protocol Exporter"
 otelJava.moduleName.set("io.opentelemetry.exporter.otlp.internal")
 
+val versions: Map<String, String> by project
 dependencies {
-  protoSource(project(":proto"))
+  protoSource("io.opentelemetry.proto:opentelemetry-proto:${versions["io.opentelemetry.proto"]}")
 
   api(project(":api:all"))
   api(project(":sdk:all"))
   api(project(":sdk:metrics"))
   compileOnly(project(":sdk-extensions:logging"))
 
-  // We only use the protos for HTTP, logging-otlp, and metric export for now. Let them expose the
-  // proto dependency in their POMs instead of here. This is fine because this artifact only
-  // contains internal code.
-  compileOnly(project(":proto"))
```

Comment:
I had forgotten to remove this before, and it was ending up putting protobuf-java on the compile classpath and we accidentally used it. Unit tests didn't catch it since our tests always rely on protobuf-java and include it on the runtime classpath. Luckily this shouldn't happen anymore with this compileOnly removed